server/server.go

```diff
@@ -1256,6 +1258,11 @@ func (s *Server) campaignLeader() {
 		return
 	}
 	s.member.EnableLeader()
+	defer func() {
+		s.member.ResetLeader()
+		// as soon as cancel the keepalive.
+		cancel()
+	}()
```

Comment:
I think the TSO resetting better should be moved to here too.

Reply:
Once leadership is reset and tso will stop service.